tests/pom.xml

```diff
@@ -1129,7 +1129,7 @@
                       <time>${service.startup.timeout}</time>
                       <http>
                         <method>GET</method>
-                        <url>http://${deviceregistry.ip}:${deviceregistry.health.port}/readiness</url>
+                        <url>http://${deviceregistry.ip}:${deviceregistry.health.port}/q/health</url>
```

Comment:
that would not cover/include the depoyment startup check.

Reply:
Ok, I see. 
In general, I wonder if it makes sense at all for an application to be ready but not started. An application that is marked as ready to serve external requests but that is not marked as started and hence not bound to liveness checks yet doesn't make much sense to me. (I haven't found any corresponding remark in the SmallRye/MicroProfile Health docs, though.)
I would expect the `started` state to be a precondition for the application to be ready.
That would also make the K8s `readinessProbe` definition more intuitive, mapping it to `/q/health/ready`.
@sophokles73 WDYT?

Concerning the Kafka connection: the actual connection creation is already decoupled from the startup check, the Kafka broker URI only has to be resolvable for startup to succeed (a check with retries is applied here).

Reply:
> the actual connection creation is already decoupled from the startup check, the Kafka broker URI only has to be resolvable for startup to succeed (a check with retries is applied here).

I know, but that is effectively the same thing because when we deploy the kafka broker along with Hono, e.g. using the Helm chart, then being able to resolve the broker host name effectively means being able to connect. So, FMPOV the DNS check could also be moved to the readiness check ...

Reply:
> That would also make the K8s readinessProbe definition more intuitive, mapping it to /q/health/ready.

Yes, and more recent versions of k8s also support a dedicated [Startup Probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes) which could invoke `/q/health/started` ...

Reply:
So, in order to use `/q/health/ready` for the readiness probe, there are still changes needed in this PR, aren't there? Adding the "started" condition to the readiness check (e.g. for the auth server).

Reply:
Why would we want/need to use `/q/health/ready` here? `/q/health` already does the trick ...

Reply:
My understanding is that for the readiness probe, one would use `/q/health/ready` - simply because for both the K8s readiness probe and for the MicroProfile Health readiness check, the definition states that this is the time when the application is ready to process requests. And it's also the intuitive mapping.
If we can't use `/q/health/ready` here (as is currently the case for the auth server), then the `/q/health/ready` output of the component is not really correct, in the sense of meaning "ready to process external requests".

Reply:
I see your point, let me think about it ...

Reply:
I guess in our case it makes sense to change the type of the DeploymentStartupCheck from `@Startup` to `@Readiness`. It will then indeed be a pre-requisite for readiness if used together with other readiness checks but it would not require other meaningful checks to be registered in order to determine if a component is really ready to be used.

Reply:
Downside would only be that the startup check would then return UP sooner than we want. I guess `@Startup` and `@Readiness` cannot be used together, can they?

Reply:
We do not have any explicit startup checks anymore in this case. Invoking the `/q/health/started` endpoint would always yield `OK`, which is basically the same behavior as not invoking the endpoint at all. What would be the purpose of marking a check with both `@Startup` and `@Readiness`?

Reply:
The purpose of doing so would be to configure that this particular check is a condition for both startup (i.e. delaying the point in time when K8s liveness checks begin) and readiness.
But as you said, this would anyway be a new feature and would also be not so important, since the startup times of Hono are not so long usually.

Reply:
Ok, so we'll keep it as a `@Readiness` check, right?

Reply:
> Ok, so we'll keep it as a `@Readiness` check, right?

Yes.